From 0ce348b4e6f1d3e2557f0471f3415e6962957bcf Mon Sep 17 00:00:00 2001
From: Noam Lessner <nlessner@habana.ai>
Date: Sun, 01 Mar 2020 08:31:09 +0200
Subject: [PATCH] [SW-8229] goya2: disable apb access in pci recovery

Because ARC is running if we don't diable the APB interface there
will be a backdoor open for jtag interface to access all PSOC via
the ARC processor. This will make sure that just AXI interface is open
via DAP and because the RR configuration only defined registers can
be accessed.

Change-Id: Iad01e33c957a6f3164dbbcb7935f1adc9ee6c674
Signed-off-by: Noam Lessner <nlessner@habana.ai>
---

diff --git a/common/security.c b/common/security.c
index d61635b..3316559 100644
--- a/common/security.c
+++ b/common/security.c
@@ -407,6 +407,12 @@
 	secuirty_dcon_if1_rr_hbw_config(spi_en);
 }
 
+static void secuirty_jtag_dap_disable(void)
+{
+	/* close dap WA jira: H5-1803 */
+	regmap_writel(1, mmPSOC_CS_TRACE_SPECIAL_GLBL_SPARE_0);
+}
+
 /* If DBG I/F were not disabled, program the RR to allow access
  * only to SPI flash and PSOC_SECURITY
  */
@@ -415,6 +421,7 @@
 	secuirty_i2c_s_rr_hbw_config(spi_en);
 	secuirty_smi_rr_hbw_config(spi_en);
 	secuirty_jtag_rr_hbw_config(spi_en);
+	secuirty_jtag_dap_disable();
 }
 
 void secuirty_i2c_s_enable(void)
